### PR TITLE
Be more accepting of ways people express InChI

### PR DESCRIPTION
**Pull request by kroenlein**
Improved filtering around InChI strings.  I think this meets the requested functionality.

### CODE REVIEW

File: gemd/entity/value/inchi_value.py [hunk 1]
```diff
@@ -22,14 +22,19 @@ def __init__(self, inchi=None):
...
         return self._inchi
 
     @inchi.setter
     def inchi(self, value: str):
+        """Set the InChI, correcting for some minor variations in format."""
         if value is None:
             self._inchi = None
         elif isinstance(value, str):
+            if not value.lower().startswith('inchi'):
+                value = f"InChI=1S/{value}"
```
Comment (lkubie) [inline gemd/entity/value/inchi_value.py:35]: I've seen this with the `1s/` so I want to check for that too.  [Here's Sigma Aldrich for example](https://www.sigmaaldrich.com/US/en/product/aldrich/757136)


```suggestion
                if not value.lower().startswith("1s/") and not value.lower().startswith("1/"):
                    value = f"InChI=1S/{value}"
                else:
                    value = f"InChI={value}"
```

File: gemd/entity/value/tests/test_inchi.py [hunk 2]
```diff
@@ -14,13 +14,16 @@ def test_json():
...
-    """Check that we can set the formula."""
+    """Check that we can set the inchi."""
     test_struct = "InChI=1/C8H8O3/c1-11-8-4-6(5-9)2-3-7(8)10/h2-5,10H,1H3"
     test_inchi = InChI()
     assert test_inchi.inchi is None
     test_inchi.inchi = test_struct
     assert test_inchi.inchi == test_struct
 
+    assert InChI("inchi=1/C8H8O3").inchi == "InChI=1/C8H8O3"
+    assert InChI("C8H8O3").inchi == "InChI=1S/C8H8O3"
```
Comment (lkubie) [inline gemd/entity/value/tests/test_inchi.py:25]:
```suggestion
    assert InChI("C8H8O3").inchi == "InChI=1S/C8H8O3"
    assert InChI("1/C8H8O3").inchi == "InChI=1/C8H8O3"
    assert InChI("1s/C8H8O3").inchi == "InChI=1s/C8H8O3"
```

**Review by lkubie — CHANGES_REQUESTED** [review]
Added a little robustness and provided a real-life example as to why.

File: gemd/entity/value/inchi_value.py (hunk 1 above)
Comment (kroenlein) [reply gemd/entity/value/inchi_value.py:35]: Great catch; thanks!

File: gemd/entity/value/tests/test_inchi.py (hunk 2 above)
Comment (kroenlein) [reply gemd/entity/value/tests/test_inchi.py:25]: Good add, though the standard expects a capital S.

**Review by lkubie — APPROVED** [review]

**Pull request merged**